Tests/TestingTests/PlanTests.swift

```diff
@@ -70,7 +73,9 @@ struct PlanTests {
     let tests = [outerTestType, deeplyNestedTest]
 
     var configuration = Configuration()
-    configuration.selectedTestIDs = [outerTestType.id, deeplyNestedTest.id]
+    configuration.testFilter = { test in
+      Test.ID.Selection(testIDs: [outerTestType.id, deeplyNestedTest.id]).contains(test)
+    }
```

Comment:
Can we avoid creating the `Test.ID.Selection` inside the filter, since as @grynspan pointed out it can cause exponential behavior. Better yet, replace all of these with `configuration.setTestFilter(toMatch:)` which is more succinct.

Reply:
ah yes! good catch